packages/material-ui-styles/src/withStyles/withStyles.d.ts

```diff
@@ -98,8 +98,11 @@ export type WithStyles<
   innerRef?: React.Ref<any> | React.RefObject<any>;
 } & PropsOfStyles<StylesType>;
 
-export interface StyledComponentProps<ClassKey extends string = string> {
-  classes?: Partial<ClassNameMap<ClassKey>>;
+export interface StyledComponentProps<ClassKey extends string | object = string> {
+  /**
+   * Override or extend the styles applied to the component.
+   */
+  classes?: ClassKey extends string ? Partial<ClassNameMap<ClassKey>> : Partial<ClassKey>;
```

Comment:
The changes here generates a lot of type errors

Reply:
What purpose does this change serve to begin with?

Reply:
Use the classes object(`AppBarClasses` etc) so that the documentation for the classes is visible in the editor

Reply:
Yeah I don't get it. Could you include a code example that makes use of this? 

I'm missing the connection between generating prop-types from type defs and this new feature.

Reply:
> I'm missing the connection between generating prop-types from type defs and this new feature.

There is none (except in-editor documentation), I just wanted to see if it would work.
I'm going to revert it